### PR TITLE
ci: Download runtime for unit tests

### PR DESCRIPTION
**Pull request by philipcmonk**
This is not trying to fix nix or ci in general, but it solves #6208.  It introduces `vere-version`, which is the runtime pace and version number to use when running the unit tests.  Generally this only needs to be updated every kelvin, but if you need a specific runtime version (eg a working branch for an upcoming kelvin release), you can add it to this file. 
 To upload a binary from non-develop/release/master branches, manually run the deployment action from urbit/vere#130.

This also deletes various things that broke and are no longer necessary, mostly because vere testing is happening in urbit/vere

### CODE REVIEW

**Comment by philipcmonk** [comment]
The tests look like they're getting cached, certainly if you touch pkg/arvo, it'll rerun the tests.

**Comment by philipcmonk** [comment]
Here's an example of it working for next/arvo, which has to pull a specific binary from urbit/vere: https://github.com/urbit/urbit/actions/runs/3953241890/jobs/6769293584

File: .github/workflows/vere.yml [hunk 1]
```diff
@@ -76,122 +74,12 @@ jobs:
...
-          exit $exitcode
-
       - if: ${{ matrix.os == 'ubuntu-latest' }}
         name: run urbit-tests
         run: |
           cp -RL tests pkg/arvo/tests
+          vere="$(cat ./vere-version | sed -e 's/\([^ ]*\) \([^ ]*\)/\1\/\2\/vere-\2/g' | tr -d '\n')"
+          url="$(echo https://bootstrap.urbit.org/vere/${vere}-linux-x86_64)"
+          curl -Lo pkg/arvo/urbit.jam "$url"
+          chmod +x pkg/arvo/urbit.jam
```
Comment (Fang-) [inline .github/workflows/vere.yml:84]: The reason for giving it `.jam` should probably be clarified in a comment so we can stop doing this as soon as it's no longer necessary.

(Also consider `vere.jam` instead, for consistency.)

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -16,21 +16,7 @@ on:
       - 'bin/**'
       - 'nix/**'
       - 'default.nix'
-  pull_request:
```
Comment (Fang-) [inline .github/workflows/build.yml:19]: Notably this actives "on" the merge of the PR branch and its target, if I understand the prior behavior and [the docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) correctly. I would expect it to still be desirable to automate running tests against the would-be end result of a merge, so unclear to me why this got removed.

**Review by Fang- — COMMENTED** [review]
A question that needs addressing and a simple suggestion, but looks good otherwise.

File: .github/workflows/build.yml (hunk 2 above)
Comment (philipcmonk) [reply .github/workflows/build.yml:19]: Ah, i didn't realize that.  Will re-add

**Review by Fang- — APPROVED** [review]
This looks good now!

**Pull request merged**